tools/automate_bindings.py

```diff
@@ -0,0 +1,215 @@
+
+TYPE_FROM_LUA_FUNCS = {
+    'int' : 'lua_tointeger',
+    'bool' : 'lua_toboolean',
+}
+
+TYPE_TO_LUA_FUNCS = {
+    'int' : 'lua_pushinteger',
+    'bool' : 'lua_pushboolean',
+}
+
+INCLUDES_TEMPLATE = r'''
+#include "boost-xtime.hpp"
+#include <Wt/%s>
+
+#include "globals.hpp"
+
+'''
+
+MODULE_FUNC_TEMPLATE = r'''
+void luawt_%(module_name)s(lua_State* L) {
+    const char* base = luawt_typeToStr<%(base)s>();
+    assert(base);
+    DECLARE_CLASS(
+        %(module_name)s,
+        L,
+        wrap<luawt_%(module_name)s_make>::func,
+        0,
+        luawt_%(module_name)s_methods,
+        %(base)s
+    );
+}
+'''
+
+RETURN_CALLS_TEMPLATE = r'''
+    %s(L, result);
+    return 1;
+'''
+
+def parse(filename):
+    # Find out the c++ parser
+    generator_path, generator_name = utils.find_xml_generator()
+    # Configure the xml generator
+    xml_generator_config = parser.xml_generator_configuration_t(
+        xml_generator_path = generator_path,
+        xml_generator = generator_name
+    )
+    # Parse the c++ file
+    decls = parser.parse([filename], xml_generator_config)
+    # Get access to the global namespace
+    global_namespace = declarations.get_global_namespace(decls)
+    return global_namespace
+
+def getMethods(global_namespace):
+    Wt = global_namespace.namespace("Wt")
+    methods = Wt.member_functions()
+    main_class = (Wt.classes())[0]
+    bases = main_class.bases
+    return methods, bases
+
+def writeSourceToFile(module_name, source):
+    with open('src/luawt/%s.cpp' % module_name, 'wt') as f:
+        f.write(source)
+
+def getModuleName(filename):
+    return os.path.basename(filename)
+
+def generateIncludes(module_name):
+    return INCLUDES_TEMPLATE.lstrip() % module_name
+
+def getSelf(module_name):
+    frame = r'''
+    %s* self = luawt_checkFromLua<%s>(L, 1);
+    '''
+    return frame % (module_name, module_name)
+
+def getInbuiltTypeArgument(options):
+    frame = r'''
+    %(argument_type)s %(argument_name)s = %(func)s(L, %(index)s);
+    '''
+    return frame.lstrip() % options
+
+def getComplexArgument(options):
+    frame = r'''
+    %(argument_type)s* %(argument_name)s =
+        luawt_checkFromLua<%(argument_type)s>(L, %(index)s);
+    '''
+    return frame.lstrip() % options
+
+def callWtFunction(return_type, args, method_name):
+    call_s = 'self->%s(' % method_name
+    args_s = ', '.join(arg.name for arg in args)
+    func_s = call_s + args_s + ');'
+    if return_type == 'void':
+        return func_s
+    else:
+        return '%s result = %s' % (return_type, func_s)
+
+def returnValue(return_type):
+    if return_type == 'void':
+        return 'return 0;'
+    else:
+        if return_type:
+            func_name = TYPE_TO_LUA_FUNCS[return_type]
+        else:
+            func_name = 'luawt_toLua'
+        return RETURN_CALLS_TEMPLATE % func_name
+
+def getInbuiltType(type):
+    for inbuilt_type in TYPE_FROM_LUA_FUNCS:
+        if type.find(inbuilt_type) is not -1:
```

Comment:
```python
if inbuilt_type not in type:
```

Can you also rename `type` to something else, because `type` is a built-in function in Python, which returns type of its argument?